### PR TITLE
tests: add fixures and basic tests for graphql functionality

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Test a basic {workflows{id,status}} query.
* Test multiple queries are properly split up and reissued.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Dammit, tests passed locally, `pip install --upgrade`ed my packages, now I get this error, investigating.

**Comment by oliver-sanders** [comment]
Fixed, the jupyterhub stuff was being imported by `--doctest-modules` which was messing with the environment causing the server to think auth was provided by the hub.

Added the hub files to the `ignore_collect` list which has to be implemented in a special way because pytest-doctest doesn't appear to read in `collect_ignore` items from the standard locations.

**Review by dwsutherland — APPROVED** [review]
I don't really understand how `jp_fetch` relates to `jp_server_config`... Some test magic/black-box tying this all together I assume... 🤯 

I do, however, understand what the new tests are testing.

The tests work locally, and coverage has increased.. 👍

**Comment by oliver-sanders** [comment]
> I don't really understand how jp_fetch relates to jp_server_config... Some test magic/black-box tying this all together I assume... exploding_head

Magic!

We are borrowing from the Jupyter Server test framework which provides a fixture called jp_serverapp which launches a new server.

https://github.com/jupyter-server/jupyter_server/blob/82cd79181f10bf6447a4262674cea9420c0140f9/jupyter_server/pytest_plugin.py#L300-L306

The `jp_fetch` fixture depends on `jp_serverapp`:

https://github.com/jupyter-server/jupyter_server/blob/3af6396d800a803eed8c71339db3eb271ed13020/jupyter_server/pytest_plugin.py#L327-L328

So whenever you see `jp_fetch` it is spinning up a server under the hood.

Note that the `jp_serverapp` fixture has function level scoping so you get a *new* server instance for every `def test`.

The Cylc integration test battery has module scoped variants of heavy-weight fixtures e.g. `mod_flow` and `mod_run` to allow multiple tests in the same module to share the same fixture.

File: cylc/uiserver/tests/test_graphql.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                pause(workflows: ["*"]) {
+                    result
+                }
+                stop(workflows: ["*"]) {
+                    result
+                }
+            }
+        ''',
+    )
+    assert response.code == 200
```
Comment (wxtim) [inline cylc/uiserver/tests/test_graphql.py:134]: consider 
```
if response.code != 200:
    raise SomePytestError(response.reason)

**Review by wxtim — APPROVED** [review]
Nothing problematic. Happy for you to merge if you wish to disregard my single comment.

File: cylc/uiserver/tests/test_graphql.py (hunk 1 above)
Comment (oliver-sanders) [reply cylc/uiserver/tests/test_graphql.py:134]: If the test fails the full response (including the `response.reason`) and server logs are dumped in the pytest output so this is already well captured.

**Pull request merged**